lib/src/model/puzzle/puzzle_controller.dart

```diff
@@ -195,12 +197,24 @@ class PuzzleController extends _$PuzzleController {
     });
   }
 
+  NormalMove solutionMove() {
```

Comment:
This can be private. And also a getter instead of a method.

Also I'd name it `nextSolutionMove` since it depends on the current move index.